lib/osf-components/addon/components/subjects/manager/component.ts

```diff
@@ -119,9 +119,10 @@ export default class SubjectManagerComponent extends Component {
         });
         this.incrementProperty('selectedSubjectsChanges');
         this.incrementProperty('savedSubjectsChanges');
+        this.model.set('subjects', savedSubjects);
         if (this.hasSubjects) {
-            const hasSubjects = savedSubjectIds.size > 0;
-            this.hasSubjects(hasSubjects);
+            this.metadataChangeset?.validate('subjects');
+            this.hasSubjects(savedSubjectIds.size > 0);
```

Comment:
If we're setting the subjects on the model, can we just do something like
```
        this.model.set('subjects', savedSubjects);
        if (this.hasSubjects) {
            this.metadataChangeset?.validate('subjects');
        }
```
And not have to deal with `this.hasSubjects` entirely? I feel like this is doing something redundant here